images/packer/packer.json

```diff
@@ -0,0 +1,103 @@
+{
+  "variables": {
+    "aws_access_key": "",
+    "aws_secret_key": "",
+    "build_version": null,
+    "kubernetes_version": "1.11.2-00",
+    "kubernetes_cni_version": "0.6.0-00",
+    "existing_ansible_ssh_args": "{{env `ANSIBLE_SSH_ARGS`}}",
+    "ami_groups": "all",
+    "ami_regions": "ap-south-1,eu-west-3,eu-west-2,eu-west-1,ap-northeast-2,ap-northeast-1,sa-east-1,ca-central-1,ap-southeast-1,ap-southeast-2,eu-central-1,us-east-1,us-east-2,us-west-1,us-west-2"
+  },
+  "builders": [
+    {
+      "name": "ubuntu-1604",
+      "type": "amazon-ebs",
+      "instance_type": "t2.small",
+      "source_ami": "{{user `ubuntu_16_04_ami`}}",
+      "ami_name": "ami-ubuntu-16.04-{{user `kubernetes_version`}}-{{timestamp}}",
+      "ami_groups": "{{user `ami_groups`}}",
+      "ami_regions": "{{user `ami_regions`}}",
+      "access_key": "{{user `aws_access_key`}}",
+      "secret_key": "{{user `aws_secret_key`}}",
+      "ssh_username": "ubuntu",
+      "tags": {
+        "build_version": "{{user `build_version`}}",
+        "source_ami": "{{user `ubuntu_16_04_ami`}}",
+        "build_date": "{{isotime}}",
+        "distribution": "Ubuntu",
+        "distribution_release": "xenial",
+        "distribution_version": "16.04",
+        "kubernetes_version": "{{user `kubernetes_version`}}",
+        "kubernetes_cni_version": "{{user `kubernetes_cni_version`}}"
+      }
+    },
+    {
+      "name": "centos-74",
+      "type": "amazon-ebs",
+      "instance_type": "t2.small",
+      "source_ami": "{{user `centos_7_4_ami`}}",
+      "ami_name": "ami-centos-7.4-{{user `kubernetes_version`}}-{{timestamp}}",
+      "ami_groups": "{{user `ami_groups`}}",
+      "ami_regions": "{{user `ami_regions`}}",
+      "access_key": "{{user `aws_access_key`}}",
+      "secret_key": "{{user `aws_secret_key`}}",
+      "ssh_username": "centos",
+      "tags": {
+        "build_version": "{{user `build_version`}}",
+        "source_ami": "{{user `centos_7_4_ami`}}",
```

Comment:
I'm wondering if we can use `source_ami_filter` here instead to be able to automatically choose the newest base image rather than having to hard code it into a variable file like we are currently doing.

Thoughts?

Reply:
Yeah. I was actually going to change it to use the official versions. The reason I left it untouched is that with the current behavior it's easier for folks to provide their own base image, which might contain different drivers/tweaks/security tools.